polus-recycle-vector-plugin/src/main.py

```diff
@@ -191,89 +38,79 @@ def main():
-    logger.info("Found the following [r][z][t] combinations:")
-    rs = [key for key in stitch_vector.keys()]
-    rs.sort()
-    for r in rs:
-        zs = [key for key in stitch_vector[r].keys()]
-        zs.sort()
-        for z in zs:
-            ts = [key for key in stitch_vector[r][z].keys()]
-            ts.sort()
-            for t in ts:
-                logger.info("[r][z][t]: [{}][{}][{}]".format(r,z,t))
-    
-    # Parse all files in the image collection directory to get position, channel, etc
-    logger.info("Parsing files in collection...")
-    collection_index = _parse_files(collection_dir,new_collection_regex,collection_variables)
-    logger.info("Found the following [r][z][t][c] combinations:")
-    rs = [key for key in stitch_vector.keys()]
-    rs.sort()
-    for r in rs:
-        zs = [key for key in collection_index[r].keys()]
-        zs.sort()
-        for z in zs:
-            ts = [key for key in collection_index[r][z].keys()]
-            ts.sort()
-            for t in ts:
-                cs = [key for key in collection_index[r][z][t].keys()]
-                cs.sort()
-                for c in cs:
-                    logger.info("[r][z][t][c]: [{}][{}][{}][{}]".format(r,z,t,c))
-    
-    # Use the r,z,t information from the existing stitching vectors to generate new stitching vectors
-    # using image names from the image collection.
-    # The if-statements below are used to catch instances where a value for a variable is present in
-    # the stitching vectors but not in the image collections. For example, if there is a vector where
-    # r=1 in the stitching vector but there is no image set with r=1, then move to the next vector.
-    logger.info("Generating stitching vectors...")
-    out_index = 1
-    rs = [key for key in stitch_vector.keys()]
-    rs.sort()
-    for r in rs:
-        if r not in collection_index.keys():
-            continue
-        zs = [z for z in stitch_vector[r].keys()]
-        zs.sort()
-        for z in zs:
-            if z not in collection_index[r].keys():
-                continue
-            ts = [t for t in stitch_vector[r][z].keys()]
-            ts.sort()
-            for t in ts:
-                if t not in collection_index[r][z].keys():
-                    continue
-                cs = [c for c in collection_index[r][z][t].keys()]
-                cs.sort()
-                for c in cs:
-                    fname = "img-global-positions-{}.txt".format(out_index)
-                    logger.info("Generating vector: {}".format(fname))
-                    fout = os.path.join(output_dir,fname)
-                    out_index += 1
-                    _generate_stitch(fout,stitch_vector[r][z][t],collection_index[r][z][t][c])
+    # Loop through the stitching vectors
+    vectors = [v for v in Path(stitch_dir).iterdir() if Path(v).name.startswith('img-global-positions')]
+    vector_count = 1
+    for vector in vectors:
```

Comment:
This is a sweet update to this codebase. More succinct and more readable simultaneously. Nice!